packages/@uppy/xhr-upload/src/index.js

```diff
@@ -287,9 +281,17 @@ module.exports = class XHRUpload extends Plugin {
           this.uploaderEvents[file.id].remove()
           this.uploaderEvents[file.id] = null
         }
+        const body = opts.getResponseData(xhr.responseText, xhr)
```

Comment:
I don't think this should be done in the `error` handler. AFAIK the response is always null here, so there is no data to retrieve.